convert2rhel/utils.py

```diff
@@ -582,4 +581,12 @@ def get_rpm_header(rpm_path, _open=open):
     return rpmhdr
 
 
+def prepare_environment():
+    """Load LCC_ALL=C to the main app process.
+
+    Use default language for output
+    """
+    os.environ.update({"LCC_ALL": "C"})
```

Comment:
No need for this when we have https://github.com/oamg/convert2rhel/blob/main/convert2rhel/utils.py#L145.

Reply:
This MR removes this line, also resolves the issue, when the main process environment was unavailable in the subprocess

Reply:
> resolves the issue, when the main process environment was unavailable in the subprocess

I believe that the issue lied in the fact that in `run_subprocess` we've been replacing all environment variables with just the LC_ALL one ([OAMG-4810](https://issues.redhat.com/browse/OAMG-4810)).

So, to solve [OAMG-4810](https://issues.redhat.com/browse/OAMG-4810) and https://github.com/oamg/convert2rhel/issues/241, this one-line change should be sufficient:
```
    process = subprocess.Popen(
        cmd,
        stdout=subprocess.PIPE,
        stderr=subprocess.STDOUT,
        bufsize=1,
        env=dict(os.environ.copy(), **{"LC_ALL": "C"}),
    )
```

Reply:
You're right. I just solved this in a bit different way. Loading this environment into the main process at the very beginning (so all children processes are inherited this environment by default). 

This os.environ update is very cool, as it is not just updating the dict, but also do os.putenv for each new env var. 

The new solution (with os.environ.update) seems cleaner to me, as we prepare the environment at the beginning of the app runtime.

What do you think?

Reply:
You variant is fine as well.

Because you just set locale, please rename the function to `set_locale()`. And the env var is not LCC_ALL, but LC_ALL.

```python
def set_locale():
    """Set the POSIX default locale for the main process as well as the child processes.

    The reason is to get predictable output from the executables we call, not influenced by non-default locale.
    """
    os.environ.update({"LC_ALL": "C"})
```